wp-graphql.php

```diff
@@ -41,13 +48,73 @@
 	require_once __DIR__ . '/src/WPGraphQL.php';
 }
 
+/**
+ * @return bool
+ */
+function graphql_can_load_plugin(): bool {
+
+	/**
+	 * WPGRAPHQL_AUTOLOAD can be set to "false" to prevent the autoloader from running.
+	 * In most cases, this is not something that should be disabled, but some environments
+	 * may bootstrap their dependencies in a global autoloader that will autoload files
+	 * before we get to this point, and requiring the autoloader again can trigger fatal errors.
+	 *
+	 * The codeception tests are an example of an environment where adding the autoloader again causes issues
+	 * so this is set to false for tests.
+	 */
+	// @phpstan-ignore-next-line: this is ignored as the constant could be defined in wp-config, prior to being defined above
+	if ( defined( 'WPGRAPHQL_AUTOLOAD' ) && false === WPGRAPHQL_AUTOLOAD ) {
```

Comment:
@jasonbahl this change is breaking for me on bedrock (and my assumption is _any_ setup using composer + normal packagist), since the autoloader isnt in `wp-graphql/vendor`.

![image](https://github.com/wp-graphql/wp-graphql/assets/29322304/80ffcb59-7a72-4b31-bbf1-abbdbe80053f)

Reply:
If we check for class_exists( `\WPGraphQL` ) here, does that fix it? 

Like if autoload is false _or_ WPGraphQL class already exists?

Reply:
Not by my desk to play around, and I'm not entirely sure the source of the bug this PR is fixing, but yeah I think the previous (intended) logical flow was more robust:

1. If `WPGraphQL` exists, we clearly have the autoloader generated which means either composer was run, it's being autoloaded somewhere else, it's dist version, or someone shimmed the class.
2. If the class doesn't, then we should try and and load the autoloader, and gracefully let people know it's missing.
3. If there's _intentionally_ no autoloader and WPGraphQL still doesn't exist, then the user should manually set a constant that tells WPGraphQL not to error out. (At least that's been my assumption what the autoload constant is for).

Reply:
Ya so it's tricky. 

We had an admin notice that warned folks that cloned the plugin or downloaded the zip from GitHub (not from the release page) that the dependencies don't exist. 

That admin notice was being called from a class, a class that didn't exist if the dependencies didn't exist, so the users we were trying to warn, never saw the warning 😂 

So, it's trying to fix that.